### PR TITLE
Update release-overview.md

### PR DESCRIPTION
**Pull request by justinegeffen**
Some terminology adjustments.

### CODE REVIEW

File: operations/research-and-development/product/release-process/release-overview.md [hunk 1]
```diff
@@ -1,33 +1,31 @@
 # Release Overview
 
-Mattermost ships with a new version on the 16th of each month in [binary form](http://docs.mattermost.com/administration/upgrade.html#mattermost-team-edition).
+Mattermost ships with a new version on the 16th of each month in [binary form](http://docs.mattermost.com/administration/upgrade.html#mattermost-team-edition) for self-hosted customers. Mattermost also ships releases with new features and bug fixes [to Mattermost Cloud](https://docs.mattermost.com/guides/cloud-admin-guide.html) every two weeks.
```
Comment (amyblais) [inline operations/research-and-development/product/release-process/release-overview.md]: The two links are now broken due to the documentation re-structuring, and I'm not 100% sure where those used to link to.

Comment (justinegeffen) [reply operations/research-and-development/product/release-process/release-overview.md]: Oh great catch, thank you. I'll fix those.

Comment (justinegeffen) [inline operations/research-and-development/product/release-process/release-overview.md]:
```suggestion
Mattermost ships with a new version on the 16th of each month in [binary form](https://docs.mattermost.com/upgrade/upgrading-mattermost-server.html#upgrading-to-the-latest-version) for self-managed customers. Mattermost also ships releases with new features and bug fixes [to Mattermost Cloud](https://docs.mattermost.com/about/product.html#mattermost-cloud) every two weeks.
```

File: operations/research-and-development/product/release-process/release-overview.md [hunk 2]
```diff
@@ -186,7 +190,7 @@ When triaging a bug report, consider the following:
 
 **Q: When is release branch cut for a Self-Managed release?**
```
Comment (justinegeffen) [inline operations/research-and-development/product/release-process/release-overview.md]:
```suggestion
**Q: When is release branch cut for a self-managed release?**
```

File: operations/research-and-development/product/release-process/release-overview.md [hunk 3]
```diff
@@ -247,7 +251,7 @@ When triaging a bug report, consider the following:
 
   - Mobile and Desktop App releases are not based off of Mattermost Cloud releases at this point.
 
-**Q: How do we track feature differences for Mattermost Cloud and Self-Managed releases?**
+**Q: How do we track feature differences for Mattermost Cloud and self-managed releases?**
 
   - Any feature that's promoted out of the feature flag process would be turned on for the Self-Managed releases. That way Self-Managed releases only get fully tested features that are ready for release, and does not get the ones still being tested. There will be some cases where we need to make a fix or change something for the Self-Managed release (e.g., a bug that only occurs if you use a certain Self-Managed setting or a certain database we don't use in Mattermost Cloud). In those cases the Self-Managed release might have some changes that are not a direct copy of the Mattermost Cloud build.
```
Comment (justinegeffen) [inline operations/research-and-development/product/release-process/release-overview.md]:
```suggestion
  - Any feature that's promoted out of the feature flag process would be turned on for the self-managed releases. That way self-managed releases only get fully tested features that are ready for release, and does not get the ones still being tested. There will be some cases where we need to make a fix or change something for the self-managed release (e.g., a bug that only occurs if you use a certain self-managed setting or a certain database we don't use in Mattermost Cloud). In those cases the self-managed release might have some changes that are not a direct copy of the Mattermost Cloud build.
```

**Review by amyblais — APPROVED** [review]
Thank you!

**Pull request merged**